File: skyspacefourtyfour_ui/src/services/LuggageService.js
import http from "../http-common"; 

class LuggageService {
  getAllLuggages(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/luggage/luggages`, searchDTO);
  }

  get(luggageId) {
    return this.getRequest(`/luggage/${luggageId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/luggage?field=${matchData}`, null);
  }

  addLuggage(data) {
    return http.post("/luggage/addLuggage", data);
  }

  update(data) {
  	return http.post("/luggage/updateLuggage", data);
  }
  
  uploadImage(data,luggageId) {
  	return http.postForm("/luggage/uploadImage/"+luggageId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new LuggageService();

File: src/main/java/com/skyspacefourtyfour/service/LuggageService.java
package com.skyspacefourtyfour.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspacefourtyfour.domain.Luggage;
import com.skyspacefourtyfour.dto.LuggageDTO;
import com.skyspacefourtyfour.dto.LuggageSearchDTO;
import com.skyspacefourtyfour.dto.LuggagePageDTO;
import com.skyspacefourtyfour.dto.LuggageConvertCriteriaDTO;
import com.skyspacefourtyfour.service.GenericService;
import com.skyspacefourtyfour.dto.common.RequestDTO;
import com.skyspacefourtyfour.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface LuggageService extends GenericService<Luggage, Integer> {

	List<Luggage> findAll();

	ResultDTO addLuggage(LuggageDTO luggageDTO, RequestDTO requestDTO);

	ResultDTO updateLuggage(LuggageDTO luggageDTO, RequestDTO requestDTO);

    Page<Luggage> getAllLuggages(Pageable pageable);

    Page<Luggage> getAllLuggages(Specification<Luggage> spec, Pageable pageable);

	ResponseEntity<LuggagePageDTO> getLuggages(LuggageSearchDTO luggageSearchDTO);
	
	List<LuggageDTO> convertLuggagesToLuggageDTOs(List<Luggage> luggages, LuggageConvertCriteriaDTO convertCriteria);

	LuggageDTO getLuggageDTOById(Integer luggageId);







}





